lib/jetstream/api/stream.ex

```diff
@@ -323,6 +323,25 @@ defmodule Jetstream.API.Stream do
     end
   end
 
+  @doc """
+  Purges some of the messages in a stream.
+
+  ## Examples
+
+      iex> Jetstream.API.Stream.create(:gnat, %Jetstream.API.Stream{name: "stream", subjects: ["sub1", "sub2"]})
+      iex> Jetstream.API.Stream.purge(:gnat, "stream", %{filter: "sub1"})
+      :ok
+
+  """
+  @spec purge(conn :: Gnat.t(), stream_name :: binary()) :: :ok | {:error, any()}
```

Comment:
I needed to add support for filtered purges in order to delete the chunks for a single object without purging the whole stream